### PR TITLE
Quad c14n clarification

### LINKED ISSUE #37: First degree hash algorithm and exact specification of hash
[§4.7.3 (5) and (6)](https://w3c.github.io/rdf-canon/spec/#hash-1d-quads-algorithm) talks about sorted nquads being joined that is then hashed. "Sort" has been discussed; what does "join" mean _exactly_? Is it identical to, say, javascript's join operation on an array of strings, ie, the lines are concatenated with no space included? Or something else? I think this should be unambigously defined.

### PR DESCRIPTION
**Pull request by gkellogg**
* Use INFRA definition of "list".
* Define, at least temporarily, "canonical n-quads form".
* Fix error in Shared hashes example, and add framework for future validation.

Fixes #37.

cc/ @afs

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/rdf-canon/pull/43.html" title="Last updated on Nov 25, 2022, 7:06 PM UTC (979593a)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/rdf-canon/43/b393900...979593a.html" title="Last updated on Nov 25, 2022, 7:06 PM UTC (979593a)">Diff</a>

### CODE REVIEW

**Review by iherman — APPROVED** [review]

**Comment by iherman** [comment]
Minor nit: spelling mistake in 'explanation' in the detail element in 4.5.2. (1)

File: spec/index.html [hunk 1]
```diff
@@ -358,6 +358,21 @@ <h2>Canonicalization Algorithm Terms</h2>
       <dt><dfn>normalized dataset</dfn></dt>
       <dd>The immutable, abstract <a>RDF dataset</a> and set of normalized
         <a>blank node identifiers</a> that are produced as output by the algorithm. A <a>normalized dataset</a> is a restriction on an <a>RDF dataset</a> where all nodes are labeled, and <a>blank nodes</a> are labeled with <a>blank node identifiers</a> consistent with running this algorithm on a base <a>RDF dataset</a>. A concrete serialization of an <a>normalized dataset</a> MUST label all <a>blank nodes</a> using these stable <a>blank node identifiers</a>.</dd>
+      <dt><dfn data-lt="canonical n-quads">canonical n-quads form</dfn></dt>
+      <dd>
+        The canonicalized representation of a quad is based on
+        <a data-cite="N-Triples#canonical-ntriples">canonical N-Triples</a>
+        defined in [[N-Triples]].
+        A represent a <a>graph name</a> in the same manner as a <a>subject</a>,
```
Comment (yamdan) [inline spec/index.html:366]: Some words might as well be inserted between "A" and "represent", if I am not mistaken:
```suggestion
        A quad in <a>canonical n-quads form</a> represents a <a>graph name</a> in the same manner as a <a>subject</a>,
```

**Review by yamdan — APPROVED** [review]
Thank you @gkellogg. I added only one small suggestion to fix a (possibly) typo.

**Comment by gkellogg** [comment]
"Spellig" :)

**Review by dlongley — APPROVED** [review]
Thanks!

**Pull request merged**